### ISSUE TITLE
ERROR! Unexpected Exception: must be type, not classobj

### ISSUE THREAD

**Issue by xubin0127** [ansible/ansible#18253]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
2.1.2.0
```
##### OS / ENVIRONMENT

Ubuntu 14.04 Python 2.7
##### SUMMARY

Run ansible appear the following error:
“ERROR! Unexpected Exception: must be type, not classobj”
##### STEPS TO REPRODUCE

```

```

##### ACTUAL RESULTS

```
ipaloma@iZ23366tv76Z:~$ ansible nginx-hz -m shell -a "ifconfig" -vvv
Using /etc/ansible/ansible.cfg as config file
ERROR! Unexpected Exception: must be type, not classobj
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible", line 92, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/dist-packages/ansible/cli/adhoc.py", line 129, in run
    inventory = Inventory(loader=loader, variable_manager=variable_manager, host_list=self.options.inventory)
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 92, in __init__
    self.parse_inventory(host_list)
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 159, in parse_inventory
    group.vars = combine_vars(group.vars, self.get_group_variables(group.name))
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 549, in get_group_variables
    self._vars_per_group[groupname] = self._get_group_variables(groupname, vault_password=vault_password)
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 567, in _get_group_variables
    vars = combine_vars(vars, self.get_group_vars(group))
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 755, in get_group_vars
    return self._get_hostgroup_vars(host=None, group=group, new_pb_basedir=new_pb_basedir, return_results=return_results)
  File "/usr/local/lib/python2.7/dist-packages/ansible/inventory/__init__.py", line 816, in _get_hostgroup_vars
    host_results = self._variable_manager.add_group_vars_file(base_path, self._loader)
  File "/usr/local/lib/python2.7/dist-packages/ansible/vars/__init__.py", line 624, in add_group_vars_file
    data = self._load_inventory_file(path, loader)
  File "/usr/local/lib/python2.7/dist-packages/ansible/vars/__init__.py", line 573, in _load_inventory_file
    data = loader.load_from_file(new_path)
  File "/usr/local/lib/python2.7/dist-packages/ansible/parsing/dataloader.py", line 114, in load_from_file
    parsed_data = self.load(data=file_data, file_name=file_name, show_content=show_content)
  File "/usr/local/lib/python2.7/dist-packages/ansible/parsing/dataloader.py", line 92, in load
    new_data = self._safe_load(in_data, file_name=file_name)
  File "/usr/local/lib/python2.7/dist-packages/ansible/parsing/dataloader.py", line 146, in _safe_load
    loader = AnsibleLoader(stream, file_name)
  File "/usr/local/lib/python2.7/dist-packages/ansible/parsing/yaml/loader.py", line 36, in __init__
    AnsibleConstructor.__init__(self, file_name=file_name)
  File "/usr/local/lib/python2.7/dist-packages/ansible/parsing/yaml/constructor.py", line 37, in __init__
    super(AnsibleConstructor, self).__init__()
TypeError: must be type, not classobj
```

**Comment by mscherer** [comment]
Can you share the inventory file ?

**Comment by xubin0127** [comment]
```
# This is the default ansible 'hosts' file.
#
# It should live in /etc/ansible/hosts
#
#   - Comments begin with the '#' character
#   - Blank lines are ignored
#   - Groups of hosts are delimited by [header] elements
#   - You can enter hostnames or ip addresses
#   - A hostname/ip can be a member of multiple groups

# Ex 1: Ungrouped hosts, specify before any group headers.

#green.example.com
#blue.example.com
#192.168.100.1
#192.168.100.10

# Ex 2: A collection of hosts belonging to the 'webservers' group

#[webservers]
#alpha.example.org
#beta.example.org
#192.168.1.100
#192.168.1.110

# If you have multiple hosts following a pattern you can specify
# them like this:

#www[001:006].example.com

# Ex 3: A collection of database servers in the 'dbservers' group

#[dbservers]
#
#db01.intranet.mydomain.net
#db02.intranet.mydomain.net
#10.25.1.56
#10.25.1.57

# Here's another example of host ranges, this time there are no
# leading 0s:

#db-[99:101]-node.example.com

[webapi-testing]
10.24.234.174

[webapi-demo]
10.47.68.47

[webapi-app]
10.24.255.181
10.26.49.7

[middleware-dev]
10.47.68.147 ansible_sudo_pass="XXXXXXXXXX"

[middleware-testing]
10.46.69.88 ansible_sudo_pass="XXXXXXXXX"

[nginx-hz]
10.47.68.5 ansible_sudo_pass="XXXXXXXXX"
```

**Comment by alikins** [comment]
I haven't been able to reproduce this with stable-2.1, stable-2.2, or devel branches.

@xubin0127 could you test this with a stable-2.2 release candidate? https://github.com/ansible/ansible/releases/tag/v2.2.0.0-0.4.rc4

**Comment by alikins** [comment]
Which python yaml packages are installed?

**Comment by alikins** [comment]
The error show is usually a subclass attempting to use super() on a base class that is not a 'new style' class. In this case, the base class is yaml.constructor.Constructor, so curious which yaml implementations are installed and which versions  (ie, what versions of PyYAML and/or pyyaml and/or python-yaml and/or libyaml are installed).

**Comment by xubin0127** [comment]
in my linux enviroment, i installed the pyyaml package with the version 3.12.

**Comment by xubin0127** [comment]
Installation of this version, the problem is still:

```
ipaloma@iZ23366tv76Z:~$ ansible --version
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']
ipaloma@iZ23366tv76Z:~$ ansible nginx-hz -m shell -a "ifconfig"
ERROR! Unexpected Exception: must be type, not classobj
to see the full traceback, use -vvv
```

**Comment by alikins** [comment]
Are there host or group vars files that could be getting read?  Maybe a non-yaml file in a vars path somewhere? 

Traceback suggests it may be attempting to load group_vars from a file and is failing, which could mean a non yaml file somewhere. 

2.2 includes some fixes for cases like that, could you try out the new releases 2.2 (https://github.com/ansible/ansible/releases/tag/v2.2.0.0-1) and see if it is still a problem?

**Comment by xubin0127** [comment]
version v2.2.0.0-1, the problem is still。
attach files for group vars files。
[group_vars.zip](https://github.com/ansible/ansible/files/566654/group_vars.zip)

**Comment by alikins** [comment]
I haven't been able to reproduce this.

Are there any custom ansible plugins involved? Particularly anything that involves python yaml.

**Comment by ansibot** [comment]
@xubin0127 ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@xubin0127 This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@xubin0127 This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@xubin0127 You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
